// (c) Nick Polyak 2023 
// License: MIT License (https://opensource.org/licenses/MIT)
//
// short overview of copyright rules:
// 1. you can use this framework in any commercial or non-commercial 
//    product as long as you retain this copyright message
// 2. Do not blame the author of this software if something goes wrong. 
// 
// Also, please, mention this software in any documentation for the 
// products that use it.

syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "google/protobuf/any.proto";
import "google/protobuf/type.proto";

package NP.Grpc.RelayServiceProto;

service RelayService
{
	rpc GetTopicStream(ShortMsg) returns (stream FullMsg) {}
	rpc PublishTopic(FullMsg) returns (ShortMsg) {}
}

message ShortMsg
{    
	google.protobuf.Timestamp msgSentTime = 1;

	string errorMessage = 2;

	google.protobuf.EnumValue topic = 3;

	string room = 4;
}

message FullMsg
{
	ShortMsg metadata = 1;

	google.protobuf.Any message = 2;
}